### ISSUE TITLE
Bug: Caching response throws AttributeError on GET: Can't pickle local object 'DependencyCleanupGroup.wrap_asgi.<locals>.wrapped'

### ISSUE THREAD

**Issue by ThinksFast** [starlite-api/starlite#899]
**Describe the bug**
When caching the response of a get request for a `Template`, I get the following exception which throws a server 500 error:

`AttributeError on GET:  Can't pickle local object 'DependencyCleanupGroup.wrap_asgi.<locals>.wrapped'`

**To Reproduce**
Here is a reproducible example:  

```
import uvicorn

from starlite import Starlite, get, Request, State, Template, Controller, TemplateConfig
from starlite.template.jinja import JinjaTemplateEngine


class FrontEnd(Controller):
    @get("/", cache=60)
    async def serveHomepage(self, request: Request, state: State) -> Template:
        return Template(
            name="homepage.html.j2",
            context={"request": request},
        )

    @get("/hello-world", cache=60)
    async def hello_world(self) -> dict[str, str]:
        """Handler function that returns a greeting dictionary."""
        return {"hello": "world"}


app = Starlite(
    route_handlers=[FrontEnd],
    template_config=TemplateConfig(
        directory="src/templates", engine=JinjaTemplateEngine
    ),
    debug=True,
)

if __name__ == "__main__":
    uvicorn.run(app, host="127.0.0.1", port=8000)


```

You can use this for the Jinja template:

```
<html>
<head>
</head>
<body>
    <h1>Homepage Header</h1>
</body>
</html>
```

**Additional context**
If you remove the cache value from the `Template` route (or set it to zero), the `get` requests are served as expected.  

In the example code, I also included a route for a non-template request, `/hello-world`, which does not throw an error.  So the problem seems to lie with `Template` responses, or some subset of responses.

**Comment by ThinksFast** [comment]
Attaching traceback: 

![image](https://user-images.githubusercontent.com/114229148/205334660-bd1f2208-1824-498b-b8d7-2444771fa3ad.png)

**Comment by Goldziher** [comment]
Have you switched python versions recently?

**Comment by ThinksFast** [comment]
Nope, I’ve been on 3.11 since the day it came out

**Comment by jtraub** [comment]
@Goldziher I think it has nothing to do with the Python version. The offending code is the fact that now response is wrapped into DependencyGroup to be able to do cleanup - [here](https://github.com/starlite-api/starlite/blob/main/starlite/routes/http.py#L167)

Obviously this can not be cached and before saving response to cache we need to unwrap it first.

**Comment by provinzkraut** [comment]
@jtraub You're right. `DependencyCleaupGroup` essentially creates an ASGI middleware here.
The reasoning so we don't have to pass the cleanup group all the way up to where the response is returned, which we need to we can consistently call the cleanup immediately before the response is returned.  I'm thinking of either making a transparent wrapper so it can be unwrapped if needed, or forego the wrapping and simply accept that we have to pass it around a bit. WDYT?

**Comment by provinzkraut** [comment]
We can do both I think. The wrapper can be made into an ASGI callable that can be pickled.

**Comment by provinzkraut** [comment]
After thinking about this, the proper way to do this seems to be calling the cleanup **before** the response is cached. This would also mean we can omit passing the cleanup group around.

**Comment by provinzkraut** [comment]
@ThinksFast I created a fix for this in #902. Can you confirm this works for you?

**Comment by ThinksFast** [comment]
Confirmed, this fixed my issue, thanks!
